src/udev/scsi_id/scsi_serial.c

```diff
@@ -361,14 +457,38 @@ static int scsi_inquiry(struct scsi_id_device *dev_scsi, int fd,
 }
 
 /* Get list of supported EVPD pages */
-static int do_scsi_page0_inquiry(struct scsi_id_device *dev_scsi, int fd,
-                                 unsigned char *buffer, unsigned len) {
-        int retval;
+static int sgio_do_scsi_page0_inquiry(struct scsi_id_device *dev_scsi, int fd,
+                                      unsigned char *buffer, unsigned len) {
+        int r;
 
         memzero(buffer, len);
-        retval = scsi_inquiry(dev_scsi, fd, 1, 0x0, buffer, len);
-        if (retval < 0)
+        r = sgio_scsi_inquiry(dev_scsi, fd, 1, 0x0, buffer, len);
+        if (r < 0)
                 return 1;
+        return process_page0_inquiry_data(dev_scsi, buffer, len);
+}
+
+static int sysfs_do_scsi_page0_inquiry(struct scsi_id_device *dev_scsi, const char *devname,
+                                       unsigned char *buffer, unsigned int len) {
+        int r;
+        int fd = -EBADF;
```

Comment:
Please use `_cleanup_close_` attribute for `fd`.

Reply:
Done